sunpy/time/time.py

```diff
@@ -126,83 +129,68 @@ def _astropy_time(time):
     return time if isinstance(time, astropy.time.Time) else astropy.time.Time(parse_time(time))
 
 
-def _parse_dt64(dt):
-    """
-    Parse a single numpy datetime64 object
-    """
-    # Convert to microsecond precision because datetime cannot handle nanoseconds
-    return dt.astype('M8[us]').astype(datetime)
-
-
 @singledispatch
 def convert_time(time_string, **kwargs):
     # default case when no type matches
     raise ValueError("'{tstr!s}' is not a valid time string!".format(tstr=time_string))
 
 
-@convert_time.register(pandas.Timestamp)
-def convert_time_pandasTimestamp(time_string, **kwargs):
-    return time_string.to_pydatetime()
-
-
 @convert_time.register(pandas.Series)
 def convert_time_pandasSeries(time_string, **kwargs):
-    if 'datetime64' in str(time_string.dtype):
-        return np.array([dt.to_pydatetime() for dt in time_string])
-    else:
-        convert_time.dispatch(object)(time_string, **kwargs)
+    return ap.Time(time_string.tolist())
 
 
 @convert_time.register(pandas.DatetimeIndex)
 def convert_time_pandasDatetimeIndex(time_string, **kwargs):
-    return time_string._mpl_repr()
+    return ap.Time(time_string.tolist())
 
 
+@convert_time.register(pandas.Timestamp)
 @convert_time.register(datetime)
 def convert_time_datetime(time_string, **kwargs):
-    return time_string
+    return ap.Time(time_string)
```

Comment:
I think that all the instances of `ap.Time(....` here should take the `**kwargs` so things from the `parse_time` api get passed into `Time`